logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -146,7 +146,12 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
 
     @Override
     public synchronized void close() throws IOException {
-        this.lock.release();
+        if (this.lock != null){
+            this.lock.release();
+            if (this.lock.channel() != null && this.lock.channel().isOpen()) {
```

Comment:
@robbavey right, that's a good point with Windows ... still, maybe the most likely reason the channel would throw here is that some other mechanism closed it already between the `if` and the `close` call. So I guess just suppressing still has value?